src/joint_constraints.rs

```diff
@@ -1,3 +1,4 @@
+use std::collections::HashMap;
```

Comment:
clippy complains that this isn't used... maybe b/c it's only used in test code? hmm...